### PR TITLE
Feat/chaos monkey

### PR DESCRIPTION
**Pull request by ViRaL95**
This includes the configuration files for chaos monkey and the source code to enable chaos monkey, and set up some configuraiton settings  on Spinnaker

### CODE REVIEW

File: src/foremast/app/create_app.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def get_accounts(self, provider='aws'):
 
         return filtered_accounts
 
-    def create_app(self):
+    def create_app(self, pipeline_configs):
```
Comment (ndcampbell) [inline src/foremast/app/create_app.py]: I think this should be a keyword argument, will be more consistent with other interfaces we have.

Also, update the doc string to include it

File: src/foremast/templates/configs/pipeline.json.j2 [hunk 2]
```diff
@@ -20,5 +20,11 @@
         "handler": "main",
         "vpc_enabled": false
     },
-    "pipeline_files": []
+    "pipeline_files": [],
+    "chaos_monkey": {
+      "enabled": false,
+      "mean_time": 5,
+      "min_time": 3,
+      "exceptions": ["prod", "prods", "prodp"]
```
Comment (ndcampbell) [inline src/foremast/templates/configs/pipeline.json.j2]: `prod*` accounts should not be in the open source templates. These should be in our private templates

File: src/foremast/templates/infrastructure/app_data.json.j2 [hunk 3]
```diff
@@ -11,13 +11,37 @@
...
+                "meanTimeBetweenKillsInWorkDays": {{ pipeline_configs.chaos_monkey.mean_time }},
+                "minTimeBetweenKillsInWorkDays": {{ pipeline_configs.chaos_monkey.min_time }},
+                "grouping": "cluster",
+                "regionsAreIndependent": true,
+                "exceptions":
+                [{% for exception in pipeline_configs.chaos_monkey.exceptions %}
+                {"account": {{ exception | tojson }}, "region": "*", "stack": "*" }
+                {% if not loop.last %},{% endif %}
+                {% endfor %}
+                {% if 'prod' not in pipeline_configs.chaos_monkey.exceptions %}
```
Comment (ndcampbell) [inline src/foremast/templates/infrastructure/app_data.json.j2]: Do not do this `prod*` exception in the public templates, only in our private templates. Other users probably do not have accounts named `prod`, `prods`, etc.

File: src/foremast/app/create_app.py (hunk 1 above)
Comment (ViRaL95) [inline src/foremast/app/create_app.py]: good catch

File: src/foremast/templates/configs/pipeline.json.j2 (hunk 2 above)
Comment (ViRaL95) [inline src/foremast/templates/configs/pipeline.json.j2]: good catch

File: src/foremast/templates/infrastructure/app_data.json.j2 (hunk 3 above)
Comment (ViRaL95) [inline src/foremast/templates/infrastructure/app_data.json.j2]: good catch

File: src/foremast/runner.py [hunk 4]
```diff
@@ -84,7 +84,7 @@ def create_app(self):
         """Create the spinnaker application."""
         utils.banner("Creating Spinnaker App")
         spinnakerapp = app.SpinnakerApp(app=self.app, email=self.email, project=self.group, repo=self.repo)
-        spinnakerapp.create_app()
+        spinnakerapp.create_app(self.configs['pipeline'])
```
Comment (ndcampbell) [inline src/foremast/runner.py]: You will need to update this to use the keyword arg too

File: src/foremast/templates/infrastructure/app_data.json.j2 [hunk 5]
```diff
@@ -11,13 +11,26 @@
...
+                {"account": {{ exception | tojson }}, "region": "*", "stack": "*" }
+                {% if not loop.last %},{% endif %}
+                {% endfor %}
+                ]
+
+            }
         },
-        "user":"foremast"
-    }{% if not loop.last %},{% endif %}
+        "user":"foremast" }{% if not loop.last %},{% endif %}
```
Comment (ndcampbell) [inline src/foremast/templates/infrastructure/app_data.json.j2]: Why did you change the formatting here? I think it was better the way it was.

File: src/foremast/runner.py (hunk 4 above)
Comment (ViRaL95) [inline src/foremast/runner.py]: another good catch

File: src/foremast/templates/infrastructure/app_data.json.j2 (hunk 5 above)
Comment (ViRaL95) [inline src/foremast/templates/infrastructure/app_data.json.j2]: agreed

File: src/foremast/templates/infrastructure/app_data.json.j2 [hunk 6]
```diff
@@ -11,13 +11,27 @@
...
+                {% if not loop.last %},{% endif %}
+                {% endfor %}
+                ]
+
+            }
         },
-        "user":"foremast"
-    }{% if not loop.last %},{% endif %}
+        "user":"foremast" 
+        }{% if not loop.last %},{% endif %}
```
Comment (ndcampbell) [inline src/foremast/templates/infrastructure/app_data.json.j2]: Still not back to original formatting :p

**Comment by sijis** [comment]
There are some pieces missing from this PR
- documentation
- update config examples (if applicable)

On a similar note, what happens if someone post this json if chaosmonkey isn't enabled/configured in Spinnaker?

**Comment by ViRaL95** [comment]
@sijis This json allows us to enable/configure ChaosMonkey on Spinnaker for us

File: src/foremast/templates/infrastructure/app_data.json.j2 (hunk 6 above)
Comment (ViRaL95) [inline src/foremast/templates/infrastructure/app_data.json.j2]: sure

**Comment by ViRaL95** [comment]
@ndcampbell formatted json

**Comment by ndcampbell** [comment]
@ViRaL95 @sijis is saying that what happens if chaosmonkey is not enabled in the user's Spinnaker? We had to configure ChaosMonkey in spinnaker, but other users may not have that. What would happen if you post this JSON to a spinnaker that doesnt have ChaosMonkey? 

Also, docs at https://github.com/gogoair/foremast/tree/master/_docs will need updated to describe this new functionality.

**Comment by ViRaL95** [comment]
@sijis It looks like if chaos monkey is disabled and the JSON is posted nothing irregular happens

**Comment by ndcampbell** [comment]
@sijis - Final comments?

File: _docs/configuration_files/pipeline_json/chaos_monkey.rest [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+Min time between terminations
+
+    | *Type*: Int
+    | *Default*: ``3``
+
+``exceptions``
+**************
+
+Accounts that Chaos MOnkey will not affect
+
```
Comment (sijis) [inline _docs/configuration_files/pipeline_json/chaos_monkey.rest]: There's a case issue with 'Monkey'.

File: _docs/configuration_files/pipeline_json/chaos_monkey.rest [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+Mean time between terminations
+
+    | *Type*: Int
+    | *Default*: ``5``
+
+``min_time``
+************
+
+Min time between terminations
```
Comment (sijis) [inline _docs/configuration_files/pipeline_json/chaos_monkey.rest]: Use the full word, minimum.

Comment (sijis) [inline _docs/configuration_files/pipeline_json/chaos_monkey.rest]: also, is this in minutes, seconds, hours, days, etc

File: _docs/configuration_files/pipeline_json/chaos_monkey.rest [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+Enable or disable Chaos Monkey
+
+    | *Type*: Boolean
+    | *Default*: ``false``
+
+``mean_time``
+*************
+
+Mean time between terminations
```
Comment (sijis) [inline _docs/configuration_files/pipeline_json/chaos_monkey.rest]: What does 'mean time' mean?

File: _docs/configuration_files/pipeline_json/chaos_monkey.rest [hunk 10]
```diff
@@ -0,0 +1,39 @@
+.. _chaos_monkey:
+
```
Comment (sijis) [inline _docs/configuration_files/pipeline_json/chaos_monkey.rest]: also, change the filename extension.

Comment (ViRaL95) [inline _docs/configuration_files/pipeline_json/chaos_monkey.rest]: This issue was resolved

**Comment by ViRaL95** [comment]
@ndcampbell check out my changes
